src/code/z_actor.c

```diff
@@ -4081,11 +4081,11 @@ EnPart* func_800358DC(Actor* actor, Vec3f* spawnPos, Vec3s* spawnRot, Vec3f* arg
     if (spawnedEnPart != NULL) {
         spawnedEnPart->actor.scale = actor->scale;
         spawnedEnPart->actor.speedXZ = arg3->x;
-        spawnedEnPart->unk_150 = arg8;
-        spawnedEnPart->unk_14C = 2;
-        spawnedEnPart->unk_14E = arg4;
-        spawnedEnPart->unk_154 = arg3->y;
-        spawnedEnPart->unk_158 = arg3->z;
+        spawnedEnPart->displayList = arg8;
+        spawnedEnPart->action = 2;
+        spawnedEnPart->delay = arg4;
+        spawnedEnPart->rotation = arg3->y;
+        spawnedEnPart->rotationSpeed = arg3->z;
```

Comment:
may be neat to create a struct for `arg3`, or just make it a `f32` array, I don't think `Vec3f` is appropriate now that we know what it's used for

You can also update `func_800358DC`'s prototype (types and names of arguments) according to your findings